src/js/background.js

```diff
@@ -235,29 +235,70 @@ Badger.prototype = {
   * add any new entries that already have a parent domain in the action_map
   * and remove any old entries that are no longer in the cookie block list
   * from the action map
+  *
+  * @param {Function} [callback] optional callback, gets success status boolean
   **/
-  updateCookieBlockList: function(){
+  updateCookieBlockList: function (callback) {
     var self = this;
-    utils.xhrRequest(constants.COOKIE_BLOCK_LIST_URL, function(err,response){
-      if(err){
-        console.error('Problem fetching cookieblock list at',
-                  constants.COOKIE_BLOCK_LIST_URL, err.status, err.message);
-        return;
+
+    if (!callback) {
+      callback = _.noop;
+    }
+
+    utils.xhrRequest(constants.COOKIE_BLOCK_LIST_URL, function (err, response) {
+      if (err) {
+        console.error(
+          "Problem fetching cookieblock list at",
+          constants.COOKIE_BLOCK_LIST_URL,
+          err.status,
+          err.message
+        );
+
+        return callback(false);
       }
-      var cookieblock_list = self.storage.getBadgerStorageObject('cookieblock_list');
-      var action_map = self.storage.getBadgerStorageObject('action_map');
 
-      var newCbDomains = _.map(response.split("\n"), function(d){ return d.trim();});
+      // handle empty response
+      if (!response.trim()) {
+        return callback(false);
+      }
+
+      var newCbDomains = _.map(response.trim().split("\n"), domain => domain.trim());
+
+      // validate the response
+      if (!_.every(newCbDomains, (domain) => {
+        // all domains must contain at least one dot
+        if (domain.indexOf('.') == -1) {
+          return false;
+        }
+
+        // validate character set
+        // TODO overly restrictive but that's OK for now, we manage the list
+        if (!/^[a-z0-9][a-z0-9.-]+[a-z]$/.test(domain)) {
+          return false;
+        }
+
+        return true;
+      })) {
+        return callback(false);
+      }
+
+      var cookieblock_list = self.storage.getBadgerStorageObject('cookieblock_list');
       var oldCbDomains = Object.keys(cookieblock_list.getItemClones());
 
       var addedDomains = _.difference(newCbDomains, oldCbDomains);
       var removedDomains = _.difference(oldCbDomains, newCbDomains);
+
       log('adding to cookie blocklist:', addedDomains);
       log('removing from cookie blocklist:', removedDomains);
 
-      // Change any removed domains back to blocked status
-      _.each(removedDomains, function(domain){
+      var action_map = self.storage.getBadgerStorageObject('action_map');
+
+      // TODO watch out for "" getting removed for example
+      // TODO https://github.com/EFForg/privacybadger/issues/1474
+      _.each(removedDomains, function (domain) {
         cookieblock_list.deleteItem(domain);
+        /*
+        // Change any removed domains back to blocked status
         if(action_map.hasItem(domain)){
```

Comment:
>I bet we could fix the sites we moved to "block" explicitly by getting our changes to the CBL and adding a migration to unblock them. I'm also curious what we've removed over time.

If we do this, we should do this in a separate ticket. It's not clear we need to do this yet.